processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -90,8 +97,13 @@ func (d *Detector) Detect(ctx context.Context) (resource pdata.Resource, schemaU
 	attr.InsertString(conventions.AttributeHostType, meta.InstanceType)
 	attr.InsertString(conventions.AttributeHostName, hostname)
 
+	client, err := d.httpClientSettings.ToClient(map[config.ComponentID]component.Extension{}, d.telemetrySettings)
```

Comment:
I think it is important to provide a working extension map to this invocation.  This can be obtained from the `component.Host` provided to the processor's `Start()` method.  A new client should probably be created then and then injected into detectors as needed, rather than injecting the client settings and creating new clients on the fly.